### PR TITLE
Map insets

### PR DESCRIPTION
**Pull request by TorsteinHonsi**
Added support for `mapView.insets` to allow rendering noncontiguous regions in map insets. Until now, this has required using pre-projected map sources.

___

### To do
 - [x] Set up the basic schema. For now, documented in `studies/map-insets/demo.js`.
 - [x] Set up the basic class/TS structure with the `MapViewInset` class.
 - [x] Apply hit test to determine what points fall within which inset.
 - [x] Translate and render the user-defined border.
 - [x] Get the composite bounds of the inset's geometries and apply translation into the target area (Look into `fitToBounds`, `MapView.setView` etc).
 - [x] Decide the best structure for `field`, `borderPath` and `geoBounds`. Currently they are all different types, though related in structure.
 - [x] Go over all options, test and create tests.
 - [x] Extend existing `fromPointToLatLon` and vice versa to use new projection and insets.
 - [x] Check what happens when zooming in the main map and in inset. Should probably work like with the legacy maps.
 - [x] Make merging of the `insets` collection work so that we can demonstrate how to override individual insets.
 - [x] Styled mode.
 - [x] API and samples
 - [x] Docs articles
 - [x] Issue: data label placement on inset areas
 - [x] @todo comments in the `MapSeries.getProjectedBounds` function

### CODE REVIEW

File: docs/maps/custom-geojson-maps.md [hunk 1]
```diff
@@ -5,7 +5,12 @@ This article contains information on creating maps from data in common mapping f
...
 
-The Highcharts Maps map collection is supplied in both SVG and GeoJSON formats, and can be edited using any compatible editor. Simple changes can be made using a text editor. [InkScape](https://inkscape.org "InkScape") is a free visual editor for SVG files. Please note that different coordinate systems are used in the SVG and GeoJSON files, mainly for optimization reasons. The map collection is also supplied as executable Javascript files. These Javascript files are identical to the GeoJSON files, except for a small amount of wrapping code.
+The Highcharts Maps map collection is supplied in both SVG, GeoJSON and TopoJSON formats, and can be edited using any compatible editor. Simple changes can be made using a text editor. [InkScape](https://inkscape.org "InkScape") is a free visual editor for SVG files. The map collection is also supplied as executable JavaScript files. These Javascript files are identical to the GeoJSON files, except for a small amount of wrapping code.
```
Comment (KacperMadej) [inline docs/maps/custom-geojson-maps.md:8]: "both" used for 3 items

**Comment by TorsteinHonsi** [comment]
@KacperMadej - some review notes:

The visual tests to a large extent makes use of recommended MapView settings from the map collection. So in order to test it live:

1. Clone the map collection to a location next to the highcharts repo and check out the [Recommended MapView PR](https://github.com/highcharts/map-collection/pull/19).
2. Hack the highcharts-utils to load the maps from the neighbour map-collection repo. In app.js, line 48:
```js
app.use('/mapdata', express.static(
  //path.dirname(require.resolve('@highcharts/map-collection/package.json')),
  //*
  path.dirname(require.resolve(path.join(
    __dirname,
    '../map-collection/Export/1.1.4'
  ))),
  // */
  { maxAge: '10m' }
));
```

**Review by KacperMadej — COMMENTED** [review]
Small things in comments.

File: ts/Maps/MapView.ts [hunk 2]
```diff
@@ -484,18 +576,167 @@ class MapView {
...
+
+                    this.setView(newCenter, void 0, true, false);
+
+                }
+
+                e.preventDefault();
+            }
+        };
+        addEvent(chart, 'pan', onPan);
+        addEvent(chart, 'touchpan', onPan);
```
[the remark this answers is not in the document]
Comment (TorsteinHonsi) [reply ts/Maps/MapView.ts:690]: Yes, agreed! Out of scope for this PR though, but would be a nice improvement.

**Pull request merged**